www/_posts/2018-08-01-future-cordova-ios-webview.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: post
+author:
+    name: Shazron Abdullah
+    url: https://twitter.com/shazron
+title:  "The Future of the iOS WebView in Apache Cordova"
+categories: news
+tags: cordova-ios uiwebview wkwebview webview roadmap
+---
+
+With the announcement of the iOS 12 beta SDK at Apple’s WWDC 2018, came the news that UIWebView, the webview originally bundled with the first iOS SDK, has been [deprecated](https://developer.apple.com/documentation/uikit/uiwebview). What this means for iOS developers is that sometime in the future, Apple will remove UIWebView from their SDK, and developers should migrate to using the WKWebView component starting right now.
+
+## WebView Engine Plugins
+
+Cordova iOS, starting with version 4, has anticipated this by moving the webview that is used by the platform into a plugin. The default webview that is used is still UIWebView, but you had the option to use WKWebView instead, with the [cordova-plugin-wkwebview-engine](https://github.com/apache/cordova-plugin-wkwebview-engine) plugin. Both the UIWebView and WKWebView webviews are plugins themselves, with the former included in the cordova-ios platform.
+
+![Figure 1]({{ site.baseurl }}/static/img/blog/2018/arch-webview-engine.png)
+
+## Cordova iOS 5 with WKWebView
+
+Starting with Cordova iOS version 5, which is a planned upcoming release, we will ship both webview plugins with the cordova-ios platform, to enable developers to test and transition users to the new WKWebView component. Cordova iOS 5 will ship with a bridge webview plugin that can switch usage of the webview plugin used at runtime. Previously, you could only choose which webview you would use at build time.
+
+![Figure 2]({{ site.baseurl }}/static/img/blog/2018/bridge-webview-engine.png)
+
+<!--more-->
+
+## Migration and Transition using the Bridge WebView Plugin
+
+The developer can switch the webview used by setting a preference, that will be read at runtime when the app starts. Developers can transition users to the new WKWebView by user choice, in their app settings -- or randomly perhaps as an A/B test. This will give the developer and their users a chance to try out the new component, and they can back out and use UIWebView if there are any problems, without needing a new app release. This is intended to be used for testing and migration purposes only -- developers should aim to support only WKWebView functionality going forward.
+
+## Cordova iOS 6, WKWebView only
```

Comment:
Do we really want to "set" this to 6? Maybe there is another major release in between?

Reply:
I need a more succinct alternative to 6, something that conveys a future major version greater than 5. I'm thinking "Cordova iOS 6 (or greater), WKWebView only"

Reply:
Future Cordova iOS release?

Reply:
Ok, I will substitute "6" for Future Release